File: assets/javascripts/discourse/services/mux-api.js
import Service from "@ember/service";
import { ajax } from "discourse/lib/ajax";

/**
 * MuxApi服务
 * 提供与Mux API的交互方法
 */
export default class MuxApi extends Service {
  /**
   * 创建直接上传
   * @returns {Promise<Object>} 上传数据，包含url和video_id
   */
  createDirectUpload() {
    return ajax("/editorjs/videos/upload", {
      type: "POST"
    });
  }

  /**
   * 获取视频状态
   * @param {string} videoId Mux视频ID
   * @returns {Promise<Object>} 视频状态数据
   */
  getVideoStatus(videoId) {
    return ajax(`/editorjs/videos/${videoId}/status`, {
      type: "GET"
    });
  }
}

File: lib/editorjs/mux_api.rb
module Editorjs
  class MuxApi
    def self.create_direct_upload
      token_id = SiteSetting.editorjs_mux_token_id
      token_secret = SiteSetting.editorjs_mux_token_secret
      mp4_support = SiteSetting.editorjs_enable_mp4_download ? "standard" : "none"
      
      url = "https://api.mux.com/video/v1/uploads"
      auth = Base64.strict_encode64("#{token_id}:#{token_secret}")
      
      body = {
        new_asset_settings: {
          playback_policy: ["public"],
          mp4_support: mp4_support,
        },
        cors_origin: Discourse.base_url,
      }
      
      response = Excon.post(
        url,
        headers: {
          "Content-Type": "application/json",
          Authorization: "Basic #{auth}",
        },
        body: body.to_json,
      )
      
      JSON.parse(response.body) if [200, 201].include?(response.status)
    end
  end
end 